ms2deepscore/MS2DeepScoreMonteCarlo.py

```diff
@@ -0,0 +1,171 @@
+from typing import List, Tuple
+import numpy as np
+from matchms import Spectrum
+from matchms.similarity.BaseSimilarity import BaseSimilarity
+from tqdm import tqdm
+
+from .vector_operations import cosine_similarity_matrix, mean_pooling, std_pooling
+from .typing import BinnedSpectrumType
+
+
+class MS2DeepScoreMonteCarlo(BaseSimilarity):
+    """Calculate MS2DeepScore ensemble similarity scores and STD between a reference
+    and a query using Monte-Carlo Dropout
+
+    Using a trained model, binned spectrums will be converted into spectrum
+    vectors using a deep neural network. The MS2DeepScoreMonteCarlo similarity is then
+    the mean of n_ensemble x n_ensemble cosine similarity score between two spectrum
```

Comment:
Is'nt it the man of n_ensemble cosine similarity scores? Instead of a matrix of n_ensemble x n_ensemble?

Reply:
It is indeed not totally intuitive... but `n_ensemble` means that `n_ensemble` embeddings are created per spectrum. And since we are interested in pair-wise scores, that means we can do `n_ensemble x n_ensemble` possible comparisons with that.